test/unit/seq_io/seq_io_reader_test.cpp

```diff
@@ -0,0 +1,286 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <algorithm>
+#include <sstream>
+
+#include <gtest/gtest.h>
+
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/test/expect_range_eq.hpp>
+#include <seqan3/test/expect_same_type.hpp>
+#include <seqan3/test/tmp_filename.hpp>
+
+#include <bio/seq_io/reader.hpp>
+#include <bio/stream/detail/fast_streambuf_iterator.hpp>
+
+inline constexpr std::string_view input =
+  R"raw(>ID1
+ACGTTTTTTTTTTTTTTT
+>ID2
+ACGTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT
+>ID3 lala
+ACGTTTA
+ACGTTTTTTTT
+
+>ID4
+ACGTTTA
+>ID5 lala
+ACGTTTA
+ACGTTTTTTTT
+)raw";
+
+inline constexpr std::string_view input_bgzipped{
+  "\x1f\x8b\x08\x04\x00\x00\x00\x00\x00\xff\x06\x00\x42\x43\x02\x00\x47\x00\xb3\xf3\x74\x31\xe4\x72\x74\x76\x0f\x41\x05"
+  "\x5c\x76\x9e\x2e\x46\x58\xc4\x29\x05\x20\x73\x8d\x15\x72\x12\x73\x12\xa1\x86\x3b\x22\x5b\xc2\x05\x92\x36\x81\xcb\x00"
+  "\x39\xa6\xb8\xd5\x02\x00\xcd\x3b\x57\x80\xba\x00\x00\x00\x1f\x8b\x08\x04\x00\x00\x00\x00\x00\xff\x06\x00\x42\x43\x02"
+  "\x00\x1b\x00\x03\x00\x00\x00\x00\x00\x00\x00\x00\x00",
+  100};
+
+TEST(seq_io_reader, concepts)
+{
+    using t = bio::seq_io::reader<>;
+    EXPECT_TRUE((std::ranges::input_range<t>));
+
+    using ct = bio::seq_io::reader<> const;
+    // not const-iterable
+    EXPECT_FALSE((std::ranges::input_range<ct>));
+}
+
+void seq_io_reader_filename_constructor(bool ext_check, auto &&... args)
+{
```

Comment:
this is something where you usually do test fixtures for..

Reply:
Hm, in some cases they are very helpful, but I don't see how turning this into a fixture would make the code shorter or more readable? What is the benefit?

I have no strong feelings about this, but in general I tend to go with more pragmatic approaches nowadays. Even I, for example, have increasing difficulty understanding many of SeqAn's unit tests, because they involved so many Macros and obsucre GoogleTest features. That's also suboptimal for something that is supposed to verify the validity of the code.
Like I said, there are situations where you save a lot of code using them (e.g. alphabet tests...), but if it is just about calling a function template? Why not just call a function template? At least everyone knows what's going on and all functionality is in one file 😁